### PR TITLE
ISPN-11206 Remove Lucene and Hibernate queries from the public API

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11206

### CODE REVIEW

**Comment by gustavonalle** [comment]
rebased

**Comment by anistor** [comment]
I see a compilation error in LocalIndexSyncStateTransferTest

**Comment by gustavonalle** [comment]
Fixed

File: query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java [hunk 1]
```diff
@@ -666,15 +666,9 @@ protected RowProcessor makeProjectionProcessor(Class<?>[] projectedTypes, Object
...
-      CacheQuery cacheQuery;
-      if (indexedQueryMode == IndexedQueryMode.BROADCAST) {
-         cacheQuery = new ClusteredCacheQueryImpl<>(luceneQuery, getSearchFactory(), asyncExecutor, cache,
-               keyTransformationHandler, classes);
-      } else {
-         cacheQuery = new CacheQueryImpl<>(luceneQuery, getSearchFactory(), cache, keyTransformationHandler,
-               timeoutExceptionFactory, classes);
-      }
-      return (CacheQuery<E>) cacheQuery;
+      return new CacheQueryImpl<>(luceneQuery, getSearchFactory(), cache, keyTransformationHandler,
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java]: So asyncExecutor and indexedQueryMode params are no longer used here. Why not remove them also from method signature? We can do that, QueryEngine is not API.

File: query/src/main/java/org/infinispan/query/impl/SearchManagerImpl.java [hunk 2]
```diff
@@ -63,10 +60,10 @@ public SearchManagerImpl(AdvancedCache<?, ?> cache) {
    }
 
    @Override
-   public <E> CacheQuery<E> getQuery(String queryString, IndexedQueryMode indexedQueryMode, Class<?>... classes) {
+   public <E> CacheQuery<E> getQuery(String queryString, IndexedQueryMode indexedQueryMode) {
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/impl/SearchManagerImpl.java]: It's a bit odd we loose the ```classes``` argument in SearcManager.getQuery but QueryEngine.buildCacheQuery still takes an array of classes

File: remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/indexing/ProtobufWrapperIndexingTest.java [hunk 3]
```diff
@@ -1,111 +0,0 @@
...
-
-      cache.put(new byte[]{1, 2, 3}, value1);
-      cache.put(new byte[]{4, 5, 6}, value2);
-
-      SearchManager sm = Search.getSearchManager(cache);
-
-      SearchIntegrator searchFactory = sm.unwrap(SearchIntegrator.class);
-      assertNotNull(searchFactory.getIndexManager(ProgrammaticSearchMappingProviderImpl.getIndexName(cache.getName())));
-
-      Query luceneQuery = sm.buildQueryBuilderForClass(ProtobufValueWrapper.class)
```
Comment (anistor) [inline remote-query/remote-query-server/src/test/java/org/infinispan/query/remote/impl/indexing/ProtobufWrapperIndexingTest.java]: This test could have been kept. Just remove lines 62-78 and we're good.

File: query/src/main/java/org/infinispan/query/impl/QueryDefinition.java [hunk 4]
```diff
@@ -98,26 +85,20 @@ private QueryEngine getQueryEngine(AdvancedCache<?, ?> cache) {
...
-            hsQuery.projection(hsQueryRequest.getProjections());
+            hsQueryRequest = queryEngine.createHsQuery(queryString, null, namedParameters);
          }
-         hsQuery.firstResult(firstResult);
-         hsQuery.maxResults(maxResults);
+         hsQuery = hsQueryRequest.getHsQuery();
+         sort = hsQueryRequest.getSort();
+         hsQuery.projection(hsQueryRequest.getProjections());
       }
+      hsQuery.firstResult(firstResult);
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/impl/QueryDefinition.java]: The old version of the code was setting firstResult and maxResults here only first time, if hsQuery == null.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/RemoteQueryDslPerfTest.java [hunk 5]
```diff
@@ -150,14 +150,12 @@ public void testEmbeddedQueryDslExecution() throws Exception {
...
-      SearchManager searchManager = org.infinispan.query.Search.getSearchManager(cache);
-      org.apache.lucene.search.Query query = searchManager.buildQueryBuilderForClass(UserHS.class).get()
-            .keyword().onField("name").matching("John1").createQuery();
+      QueryFactory queryFactory = Search.getQueryFactory(cache);
+      Query cacheQuery = queryFactory.create(String.format("From %s where name = 'John1'", UserHS.class.getName()));
 
       final int loops = 100000;
       final long startTs = System.nanoTime();
       for (int i = 0; i < loops; i++) {
-         CacheQuery<User> cacheQuery = searchManager.getQuery(query);
```
Comment (anistor) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/RemoteQueryDslPerfTest.java]: Extracting this out of the loop is an optimization that impacts the measurement. I believe we want to measure total query execution time including any initial preparation/parsing time for the query. The system is free to eventually internally optimize that by using caching for parsed queries, but we should not help here.

File: query/src/main/java/org/infinispan/query/CacheQuery.java [hunk 6]
```diff
@@ -18,7 +19,7 @@
  * @author Navin Surtani
  * @author Sanne Grinovero &lt;sanne@hibernate.org&gt; (C) 2011 Red Hat Inc.
  * @author Marko Luksa
- * @see org.infinispan.query.impl.SearchManagerImpl#getQuery(org.apache.lucene.search.Query, Class...)
+ * @see org.infinispan.query.impl.SearchManagerImpl#getQuery(String, IndexedQueryMode, Class...)
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/CacheQuery.java]: This javadoc ref is broken

File: query/src/test/java/org/infinispan/query/tx/NonLocalIndexingTest.java [hunk 7]
```diff
@@ -61,7 +59,7 @@ public void testQueryAfterAddingNewNode() throws Exception {
       store("Astronaut", new AnotherGrassEater("Astronaut", "is having a hard time to find grass"), cache(1));
       //replacing same key with a new type:
       assertFind("cat", 0);
-      assertFind("grass", 1);
+      assertFind("grass", 0);
```
Comment (anistor) [inline query/src/test/java/org/infinispan/query/tx/NonLocalIndexingTest.java]: I'm confused by this. Why 0?

Comment (gustavonalle) [reply query/src/test/java/org/infinispan/query/tx/NonLocalIndexingTest.java]: The API that worked on Lucene queries would do a query targeting all entities in the cache (Person and AnotherGrassEater), with my changes it only targets the Person entity

**Comment by anistor** [comment]
Since we cannot target multiple entities could we wipe out that classes arg entirely, not just from SearchManager but also from QueryEngine and all underlying stuff ?

**Comment by gustavonalle** [comment]
> could we wipe out that classes arg entirely,

I think I tried that, but HSearch requires it at a certain point, I'll take a look again

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/RemoteQueryDslPerfTest.java (hunk 5 above)
Comment (gustavonalle) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/RemoteQueryDslPerfTest.java]: ok, I'll move it back

**Comment by gustavonalle** [comment]
Reviews addressed

**Comment by anistor** [comment]
Merged in master. Thanks @gustavonalle !

**Comment by anistor** [comment]
I see jira fix version includes  10.1.4.Final also. Do we plan to backport these changes to 10.1.x @gustavonalle , @tristantarrant ?  IMHO they are pretty safe.

**Comment by gustavonalle** [comment]
I can backport it if needed, didn't do it because of I though there'd be a code freeze in place?

**Comment by anistor** [comment]
Given the current climate changes 'freeze' is a relative term.

**Comment by anistor** [comment]
Or can we at least make a PR for 10.1.x to deprecate all things that are removed in this PR with a clear deprecation note saying they are gone without replacement in 11?  That would probably be more appropriate.

**Comment by gustavonalle** [comment]
+1

**Comment by gustavonalle** [comment]
https://github.com/infinispan/infinispan/pull/7999

**Pull request merged**